### PR TITLE
Added readme

### LINKED ISSUE #206: Add a README in accordance with NCEPLIBS requirements
I would like to add a README to match those on the other NCEPLIBS libraries. See https://github.com/NOAA-EMC/NCEPLIBS/issues/120

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #206 

Add a README similar to those used with other NCEPLIBS packages.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
Will the building instruction in  README  for building a upp lib (a library under NCEPLIBS) only  or  include building executable?

**Comment by edwardhartnett** [comment]
@WenMeng-NOAA how about both sets of build instructions. 

What is the difference between the two builds? The use of the cmake options?

**Comment by WenMeng-NOAA** [comment]
For GNC make option, upp lib building uses sorc/ncep_post.fd/build_upp_lib.sh and executable ncep_post building uses 
sorc/build_ncep_post.sh. I have no idea of scenario of cmake option. I would defer the question to @fossell . The DTC UPP team works for UPP public release for UFS.

**Comment by fossell** [comment]
@edwardhartnett - The UPP package serves several applications at this time with differing build and use mechanisms. We anticipate further unification of this in the future but at this time we're in a period of transition. Right now, the builds/uses include:
1) Operational - use GNC build as Wen described for both library and executable (library used for GFS only at this time)
2) MRW App - uses UPP packaged with nceplibs and cmake to build/run with executable (via release/public-v1 branch)
3) SRW App - uses UPP repo branch/tag directly and uses cmake to build/run with executable (via release/public-v2 branch)
4) Community standalone - uses UPP repo branch/tag directly and uses cmake to build/run with executable (via release/public-v2 branch)
     For these procedures, we add a -DCMAKE_PREFIX_PATH=${INSTALL_PREFIX} where INSTALL_PREFIX is the location of the nceplibs installation as a dependency requirement.  
5) Other ? - Users trying to use UPP library as part of nceplibs with cmake - our team is not familiar with this yet

Adding a README is on my list, so thank you for initiating this PR with requirements from nceplibs and cmake build. 
I think we can start with what you have and add to it to be more comprehensive documentation about the scope of this package.  If there is a preferred way to incorporate these other variations of build/use into the readme that compliments the cmake and nceplibs descriptions, please advise.  

@WenMeng-NOAA do you want operational build instructions in the README as well?

File: README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+This package requires the following NCEPLIBS packages:
+
+- [NCEPLIBS-g2](https://github.com/NOAA-EMC/NCEPLIBS-g2)
+- [NCEPLIBS-g2tmpl](https://github.com/NOAA-EMC/NCEPLIBS-g2tmpl)
+- [NCEPLIBS-sp](https://github.com/NOAA-EMC/NCEPLIBS-sp)
+- [NCEPLIBS-ip](https://github.com/NOAA-EMC/NCEPLIBS-ip)
+- [NCEPLIBS-bacio](https://github.com/NOAA-EMC/NCEPLIBS-bacio)
+- [NCEPLIBS-w3emc](https://github.com/NOAA-EMC/NCEPLIBS-w3emc)
+- CRTM
```
Comment (fossell) [inline README.md]: This is an important one for many of our community users. can we add repo info in some manner here too. Assume it's https://github.com/noaa-emc/emc_crtm?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- [NCEPLIBS-sfcio](https://github.com/NOAA-EMC/NCEPLIBS-sfcio) -
+- [NCEPLIBS-nemsio](https://github.com/NOAA-EMC/NCEPLIBS-nemsio) -
+- [NCEPLIBS-gfsio](https://github.com/NOAA-EMC/NCEPLIBS-gfsio)
+
+
+The [NCEPLIBS-wrf_io](https://github.com/NOAA-EMC/NCEPLIBS-wrf_io)
+library is required to build with NCEPpost with WRF-IO library (cmake
+option BUILD_WITH_WRFIO).
+
+The following third-party libraries are required:
```
Comment (fossell) [inline README.md:44]: What about jasper, zlib, and png?  I'm just getting familiar with nceplibs package and the cmake build, so apologies if this is obvious.  These are taken care of through nceplibs-external, correct? So do they not need to be listed here if using nceplibs as a dependency for cmake build of upp?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- [netcdf-c](https://github.com/Unidata/netcdf-c)
+- [netcdf-fortran](https://github.com/Unidata/netcdf-fortran)
+
+## Building
+
+```
+mkdir build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=/path/to/install 
+make -j2
```
Comment (fossell) [inline README.md]: What does `make -j2` do?  We don't have this in our building UPP executable instructions.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+
+# EMC_post
+
+The EMC_post package provides tools for post processing for NCEP models.
+
+This is part of the [NCEPLIBS](https://github.com/NOAA-EMC/NCEPLIBS)
```
Comment (fossell) [inline README.md:8]: It also serves as a standalone package that is distributed to the community. And is built outside of nceplibs for some UFS applications (e.g. SRW v1.0.0).  Should clarify this somehow?

File: README.md (hunk 1 above)
Comment (edwardhartnett) [reply README.md]: Yes, I will add.

File: README.md (hunk 4 above)
Comment (WenMeng-NOAA) [reply README.md:8]: @fossell I agree with you.

File: README.md (hunk 2 above)
Comment (edwardhartnett) [reply README.md:44]: jasper, zlib, and png are not checked for in CMakeLists.txt. Are they required to build EMC_POST? If so, they should be listed here. Also they should probably be checked for in the top-level CMakeLists.txt.

File: README.md (hunk 3 above)
Comment (edwardhartnett) [reply README.md]: It builds in parallel. It's not necessary and I will remove it.

File: README.md (hunk 4 above)
Comment (edwardhartnett) [reply README.md:8]: I have clarified.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+project. It also serves as a standalone package that is distributed to
+the community. It is built outside of nceplibs for some UFS
+applications (e.g. SRW v1.0.0).
+
+## Authors
+
+NCEP/EMC Developers
+
+Code Manager: WenMeng
```
Comment (WenMeng-NOAA) [inline README.md]: Please add the changes as
Code Managers: Wen Meng, Huiya Chuang, Kate Fossell

Comment (edwardhartnett) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (fossell) [reply README.md:44]: Actually good point. They have been requirements for so long it is engrained in my head.  But we do not require them for community build as of the most recent release. 
@WenMeng-NOAA - Can you comment/confirm the requirement of jasper, zlib, png for any operational needs?

Comment (WenMeng-NOAA) [reply README.md:44]: Yes, these three are required for operational implementation.

Comment (WenMeng-NOAA) [reply README.md:44]: I might think w3nco is required too.

Comment (edwardhartnett) [reply README.md:44]: OK, so should these be listed as requirements?

If so, should they be checked for in the CMake build?

**Comment by edwardhartnett** [comment]
Also, can you explain the different names? Is it EMC_post or UPP_post? I believe EMC_post is the historical name and UPP_post is the new name? Is this correct?

**Comment by WenMeng-NOAA** [comment]
The EMC_post is the current UPP repository name. We might have a name change in the future.

**Comment by edwardhartnett** [comment]
OK, I added UPP to the README. Does it stand for Unified Post Processing?

Take a look at the top of the README and tell me if I got it right...

File: README.md (hunk 2 above)
Comment (WenMeng-NOAA) [reply README.md:44]: Yes.

**Comment by WenMeng-NOAA** [comment]
That's right. Now the README looks good to me. I might wait for Kate's comment

File: README.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+## Prerequisites
+
+This package requires the following NCEPLIBS packages:
+
+- [NCEPLIBS-g2](https://github.com/NOAA-EMC/NCEPLIBS-g2)
+- [NCEPLIBS-g2tmpl](https://github.com/NOAA-EMC/NCEPLIBS-g2tmpl)
+- [NCEPLIBS-sp](https://github.com/NOAA-EMC/NCEPLIBS-sp)
+- [NCEPLIBS-ip](https://github.com/NOAA-EMC/NCEPLIBS-ip)
+- [NCEPLIBS-bacio](https://github.com/NOAA-EMC/NCEPLIBS-bacio)
+- [NCEPLIBS-w3emc](https://github.com/NOAA-EMC/NCEPLIBS-w3emc)
```
Comment (WenMeng-NOAA) [inline README.md:28]: A new line might be needed:
[NCEPLIBS-w3nco](https://github.com/NOAA-EMC/NCEPLIBS-w3nco)

Comment (edwardhartnett) [reply README.md:28]: Added.

**Review by WenMeng-NOAA — APPROVED** [review]
looks good to me.

**Review by fossell — APPROVED** [review]
Looks good for now.  I'd like to clean up / clarify the various build/use descriptions, but that's not @edwardhartnett responsibility. I will do that in a separate PR.

**Comment by WenMeng-NOAA** [comment]
@edwardhartnett and @fossell Thanks for your efforts for initialing README.md. I will integrate it in develop. We might expend more in the future.

**Pull request merged**